packages/react-tooltip/src/components/Tooltip/useTooltip.tsx

```diff
@@ -30,52 +45,117 @@ export const useTooltip = (
   ref: React.Ref<HTMLElement>,
   defaultProps?: TooltipProps,
 ): TooltipState => {
-  const [target, setTarget] = React.useState<HTMLElement | null>(null);
+  const [visible, setVisible] = React.useState(false);
 
   const state = mergeProps(
     {
       ref,
-      as: 'div',
+      children: props.children,
+      content: {
+        as: React.Fragment,
+      },
+      id: useId('tooltip-'),
       role: 'tooltip',
       position: 'above',
       align: 'center',
       offset: 4,
-      arrow: { as: 'div', children: '' },
-      visible: !!target,
+      showDelay: 250,
+      hideDelay: 250,
+      visible,
+      rendered: !useIsSSR(),
     },
     defaultProps && resolveShorthandProps(defaultProps, tooltipShorthandProps),
     resolveShorthandProps(props, tooltipShorthandProps),
   );
 
-  if (state.noArrow) {
-    state.arrow = undefined;
-  }
-
-  React.useImperativeHandle(
-    state.componentRef,
-    () => ({
-      show: t => setTarget(t),
-      hide: () => setTarget(null),
-    }),
-    [],
-  );
-
   const theme = useTheme();
 
-  const { targetRef, containerRef, arrowRef } = usePopper({
+  const popper = usePopper({
     enabled: state.visible,
     position: state.position,
     align: state.align,
     offset: [0, state.offset + (state.noArrow ? 0 : arrowHeight)],
     arrowPadding: theme?.global ? 2 * parseInt(tooltipBorderRadius(theme), 10) : 0,
   });
 
-  targetRef.current = target;
-  state.ref = useMergedRefs(state.ref, containerRef);
+  state.ref = useMergedRefs(state.ref, popper.containerRef);
+  state.arrowRef = popper.arrowRef;
+
+  const tooltipManager = useTooltipManager(createTooltipManager);
+
+  // Notify the manager when the pointer enters or leaves the tooltip
+  state.onPointerEnter = mergeCallbacks(() => tooltipManager.notifyEnterTooltip(), state.onPointerEnter);
+  state.onPointerLeave = mergeCallbacks(() => tooltipManager.notifyLeaveTooltip(), state.onPointerLeave);
+
+  const onEnterTrigger = (ev: React.SyntheticEvent<HTMLElement>) => {
+    const target = state.targetRef?.current ?? ev.currentTarget;
+    popper.targetRef.current = target;
+
+    if (state.onlyIfTruncated) {
+      // For tooltips that only show when truncated, don't show if the target's scroll size <= client size
+      if (target.scrollWidth <= target.clientWidth && target.scrollHeight <= target.clientHeight) {
+        return;
+      }
+    }
+
+    tooltipManager.notifyEnterTrigger({
+      setVisible,
+      trigger: ev.currentTarget,
+      showDelay: state.showDelay,
+      hideDelay: state.hideDelay,
+    });
+  };
+
+  const onLeaveTrigger = (ev: React.SyntheticEvent<HTMLElement>) => {
+    tooltipManager.notifyLeaveTrigger(ev.currentTarget);
+  };
+
+  // The aria-* prop to be set on the trigger element (if any)
+  const triggerAriaProps: TooltipTriggerProps = {};
```

Comment:
nit
```suggestion
  const triggerAriaProps: Pick<TooltipTriggerProps, 'aria-describedby' | 'aria-labelledby' | 'aria-label'> = {};
```

Reply:
(This no longer applies after reworking the code to memoize the callback functions)